### ISSUE TITLE
“E2507 IAM Policy Documents needs to be JSON” reported for JSON IAM Policy documents

### ISSUE THREAD

**Issue by acdha** [awslabs/cfn-python-lint#383]
*cfn-lint version: (`cfn-lint --version`)*

cfn-lint 0.7.3

*Description of issue.*

I have some CloudFormation templates which are written in YAML but contain various JSON IAM policies. `cfn-lint` reports these errors for the files in a working branch listed below:

```
E2507 IAM Policy Documents needs to be JSON
cloudformation/services/concordia-service/service.yaml:183:19

E2507 IAM Policy Documents needs to be JSON
cloudformation/services/concordia-service/service.yaml:224:19
```

https://github.com/LibraryOfCongress/concordia/blob/eba167e1bccf4cdd10bda010f6456095061f7dc3/cloudformation/services/concordia-service/service.yaml#L183-L204

https://github.com/LibraryOfCongress/concordia/blob/eba167e1bccf4cdd10bda010f6456095061f7dc3/cloudformation/services/concordia-service/service.yaml#L224-L244

It's not clear to me what is going on since the JSON is valid and the IAM policy validator doesn't report any errors.

**Comment by acdha** [comment]
Looking at https://github.com/awslabs/cfn-python-lint/blob/7e04573a9135aec1aa575e0679a802dfdd90178d/src/cfnlint/rules/resources/iam/Policy.py#L68 suggests that this is simply about expecting the policy document to be the same format as the template, which is stricter than the [official documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-policy.html):

> AWS Identity and Access Management (IAM) requires that policies be in JSON format. However, for templates formatted in YAML, you can create an IAM policy in either JSON or YAML format. AWS CloudFormation always converts a policy to JSON format before submitting it to IAM.

**Comment by cmmeyer** [comment]
Makes sense, and that's a common pattern since the policies you get from the IAM tooling are always JSON.  Is it ok if I mark this as an enhancement to allow mixing syntax?

**Comment by acdha** [comment]
Sounds good to me – I ended up converting it to YAML but it'd be really nice to be able to copy-and-paste a known-working policy even if the syntax is a bit odd

**Comment by kddejong** [comment]
We are now parsing the strings and testing them after being parsed.
